src/python/WMCore/WMSpec/WMWorkloadTools.py

```diff
@@ -215,7 +215,7 @@ def validateSiteLists(arguments):
         whiteList = [whiteList]
     if not isinstance(blackList, list):
         blackList = [blackList]
-    res = (set(whiteList) & set(blackList))
+    res = [s for s in list(set(whiteList) & set(blackList)) if s]
```

Comment:
Eric, it's not about empty list, it is about the case when lists contains empty
strings, e.g. whiteList=[''] and blackList=[''].

The result of set operations would be set(['']) and which is a list with single
element and that element is empty string. Once you take a len(set['']) it gives
1. The logic below which check len(res) is wrong because both whiteList and
blackList contain an empty string ''.

On  0, Eric Vaandering <notifications@github.com> wrote:
> > @@ -215,7 +215,7 @@ def validateSiteLists(arguments):
> >          whiteList = [whiteList]
> >      if not isinstance(blackList, list):
> >          blackList = [blackList]
> > -    res = (set(whiteList) & set(blackList))
> > +    res = [s for s in list(set(whiteList) & set(blackList)) if s]
> 
> what does list(set & set) give you if its empty? 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/6603/files#r53479116